### PR TITLE
code refactor + specify requirement versions!!!!!!!!!!!!!!!!!

### PR DESCRIPTION
**Pull request by CodeByAidan**
you dumbas

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -39,22 +39,24 @@ def hasNitro(self):
...
             print(sex.text)
         if len(self.sub_ids) == 0:
-            log(f"{colorama.Fore.RED}Token has no nitro, removing.")
-            self.removeTokenFromTxt()
-            return False
+            return self._extracted_from_hasNitro_7('Token has no nitro, removing.')
         log(f"{colorama.Fore.GREEN}Token has nitro.")
         return True
 
+    # TODO Rename this here and in `hasNitro`
```
Comment (livxy) [inline main.py:54]: IMPORTANT!!!!!!!!!

**Comment by qoft** [comment]
dawg u legit just made another function and renamed a few things. this doesn't fix anything

File: main.py (hunk 1 above)
Comment (livxy) [reply main.py:54]: Function ``Nitro.hasNitro`` refactored with the following changes:

- Extract duplicate code into method

```diff
@@ -4,9 +4,7 @@
             headers=self.headers,
         )
         if sex.status_code in [403, 401]:
-            log(f"{colorama.Fore.RED}Token is invalid, removing.")
-            self.removeTokenFromTxt()
-            return False
+            return self._extracted_from_hasNitro_7('Token is invalid, removing.')
         try:
             for sub in sex.json():
                 self.sub_ids.append(sub["id"])
@@ -14,8 +12,6 @@
             print(e)
             print(sex.text)
         if len(self.sub_ids) == 0:
-            log(f"{colorama.Fore.RED}Token has no nitro, removing.")
-            self.removeTokenFromTxt()
-            return False
+            return self._extracted_from_hasNitro_7('Token has no nitro, removing.')
         log(f"{colorama.Fore.GREEN}Token has nitro.")
         return True

+
+    # TODO Rename this here and in `hasNitro`
+    def _extracted_from_hasNitro_7(self, arg0):
+        log(f"{colorama.Fore.RED}{arg0}")
+        self.removeTokenFromTxt()
+        return False
```

**Comment by timof121** [comment]
no kys no version requirement number

**Comment by livxy** [comment]
💀 he merged it

**Comment by MayoneysPh** [comment]
What is Guild ID:????

**Pull request merged**